cfpb/sbl-frontend
# [Institution + Filing] Handle incomplete Institution data from API

Close #743 

The Institution API allows for the creation of Institutions with incomplete details.  This can create issues on the Frontend:
- Crashes when attempting to access non-existent nested structures 
- Form validation issues due to unexpected `null` values

## Changes
- Authenticated landing
  - Review institutions - Link to Institution's profile even if `name` is blank
- View Institution
  - Display "Not available" for `null` or empty string values
- Update institution
  - Update [formTypes.ts](https://github.com/cfpb/sbl-frontend/compare/743-bug--incomplete-institution-data?expand=1#diff-509becc91333f008e8e35c65f41489edcc124e6898ac625f544d4b5c9d552a8e) to expect these "nullable" fields, which clears the validation issues I was seeing
  - Update [processRssdId](https://github.com/cfpb/sbl-frontend/compare/743-bug--incomplete-institution-data?expand=1#diff-cfe8455ba1d3c34934f4c0e21d8a9e7a1ecd2768e89aa2a2e17b1a52a30d29bf) to account for `null` values
- Filing
  - Display the Institution's LEI in place of it's name when `institution.name` is an empty string

## How to test this PR

1. Create an Institution with minimal details
```
{
  "name": "",
  "lei": "123456789TESTBANK999",
  "is_active": true,
  "sbl_institution_types": [],
  "hq_address_street_1": "",
  "hq_address_city": "",
  "hq_address_state_code": "GA",
  "hq_address_zip": ""  
}
```
2. Associate the institution with your account
3. View the Institution profile and confirm page doesn't crash
4. Go to Update Institution profile and confirm page doesn't crash
5. Try to submit a change to the Institution profile and confirm there are no unexpected validation errors blocking form submission

## Screenshots
### Review institutions (authenticated landing) - Show profile link even when missing name
<img width="706" alt="Screenshot 2024-06-21 at 4 49 27 PM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/1667e700-db0f-4fbd-9c18-00520fed7def">

### View Institution profile - lots of missing details rendered as "Not available"
![screencapture-localhost-8899-institution-123456789MEISBANK808-2024-06-20-16_28_38](https://github.com/cfpb/sbl-frontend/assets/2592907/243c3b09-1cd8-4bc9-91ce-0d69a7564e47)

### Update Institution profile - render missing data as "Not available"
<img width="803" alt="Screenshot 2024-06-21 at 4 53 53 PM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/08bdcaf5-9086-4fb6-a43b-d16aa5eae0a7">


## Questions
- API: Who has access to create Institutions? Are these scenarios of minimal details likely?
- Review Institutions: Would we prefer the old way of forcing users to contact SBL Help when the Institution name is missing?
